src/error.jl

```diff
@@ -4,6 +4,36 @@
 #
 ###############################################################################
 
+mutable struct ImpossibleInverse{T, S} <: Exception
+  data::T   # element that could not be inverted
+  mod::S    # ring or modulus with respect to which it could not be inverted
+end
+
+function ImpossibleInverse(x::T, y::S) where {T <: RingElement, S}
+  if iszero(x)
+    throw(DivideError())
+  else
+    throw(ImpossibleInverse{T, S}(x, y))
```

Comment:
Do you really want to *throw* here? Isn't that a bit unusual in a (fake) constructor? 

The function in line 12 calls this, so it also always throws. Yet in your change in `Residue.jl`, you write `throw(ImpossibleInverse(a))` -- I think the outer `throw` there never can be executed as we throw before. Or am I missing something?

I find this a bit odd...

Reply:
Yeah, I messed something up here. I meant for the fake contructor to *return* a DivideError or *return* a real ImpossibleInverse.